### PR TITLE
Refactor project into separate libraries

### PR DESCRIPTION
**Pull request by adamkewley**
**PROTOTYPE**: Based on discussion in #562 . 

One of the most valuable things Falltergeist offers is a collection of working open-source parsers for Fallout 1/2's data files. The documentation for those formats is scattered (e.g. [on wikis](http://falloutmods.wikia.com/wiki/Category:Fallout_and_Fallout_2_file_formats), [peoples pet projects](https://github.com/adamkewley/fo2dat)) and not verified. Being able to use those parsers independently of the Falltergeist engine will enable developers to write conversion utilities and write their own game engines.

I had a few hours to spare, so I refactored Falltergeist/develop into:

- Core game engine executable (as previous)
- Data format shared library (new)
- Static library containing classes shared by the above (`falltergeist-common-cpp`)

With this project structure, you should be able to build the separate components in one step (if the IDE supports CMake), rather than needing to maintain two separate repos. With further refactoring, cleanup, and documentation we could publish the IO formats as a library that other developers can use independently of the Falltergeist game engine. This will enable developers to write, for example, command line interfaces for converting between MAP and grid files without needing to duplicate code from the falltergeist executable.

@alexeevdv , if you get the chance, could you check this PR out (or my fork) and see what you think about it. This represents a major change in the project structure, and IDEs might not play ball with it. Please comment when you get the chance.

### CODE REVIEW

**Comment by alexeevdv** [comment]
742 changes.... jesus....it will take forever to review this :)

File: falltergeist/src/Game/Game.cpp [hunk 1]
```diff
@@ -32,7 +32,7 @@
 #include "../Event/Dispatcher.h"
 #include "../Event/State.h"
 #include "../Exception.h"
-#include "../Format/Gam/File.h"
+#include <falltergeist/io//Gam/File.h>
```
Comment (alexeevdv) [inline falltergeist/src/Game/Game.cpp]: does double slash put in these includes for a reason?

Comment (adamkewley) [reply falltergeist/src/Game/Game.cpp]: That's because I had to manually sed some replacements in. It's a bug :(

**Comment by adamkewley** [comment]
I can try to move them into a separate repo, sure =) The main idea behind keeping it in the mainline was so that you can change the IO code in-branch without needing to pull releases from elsewhere

**Comment by adamkewley** [comment]
@alexeevdv I'd close this PR for now. Now that I know what you want I'll get to work on developing a separate project.

**Comment by alexeevdv** [comment]
Thanks. Please take a look into libfalltergeist source code. Maybe you'll find a way to fix raw pointers issues there

**Pull request closed without merging**